DataDog/datadog-agent
# Use pkg/forwarder for submitting process-agent payloads

### What does this PR do?

This commit modifies the process-agent to use the pkg/forwarder facility for sending HTTP payloads.  This will ensure that payloads are retried until they are successful.

One difference is that this will submit the API key in a different header than we are currently using.  The intake API will be modified to read this header.

We're also sending a new header, X-DD-Agent-Timestamp that indicates the time that we attempted to send the payload

### Motivation

What inspired you to submit this pull request?

### Additional Notes

QA:

To QA, we need to make sure that process, container, pod and connection messages are all correctly sent to intake with. and without an HTTP proxy

Additionally, we want make sure that real time mode is toggled on and off appropriately
